### PR TITLE
trigger new events for age-up

### PR DESCRIPTION
**Pull request by yufengwng**
## Context and the big picture
 
Now that we have the new event types, this PR goes back and actually triggers them.
 
 ## What type of change is this?
  
  - [ ] Bugfix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] Experimental proof-of-concept
  - [ ] Infosec/Compliance remediation
  
 ## Risk Assessment
  - [ ] These changes are **HIGH RISK**.
    - [ ] There is an elevated risk that may impact a large number of features
    - [ ] There is an elevated risk to security and privacy
    - [ ] These changes introduce new 3rd party dependencies, alter the network perimeter, or contain
    major upgrades to persistent storage or SaaS dependencies such as auth0, sendgrid, or easypost    
  - [x] These changes are low risk, do not impact security/privacy, do not impact pepper shared
  components/modules, and have no major upgrades to 3rd party services or libraries
 
 ### FUD Score 

  - [x] :relaxed: All good, business as usual!
  - [ ] :sweat_smile: There might be some issues here
  - [ ] :scream:I'm sweaty and nervous
 
 ## How to we demo these changes?

 - [x] They are user-visible in dev as a regular user journey and require no additional instructions.
 - [ ] Getting dev into a state where this is user-visible requires some tech fiddling.  I have
 documented these steps in the related ticket.
 - [ ] Requires other features before it's human visible.  I have documented the blocking issues
 in jira.
 - [ ] I have no idea how to demo this.  Please help me!
 
 ## UI/UX
  - [x] There ain't no UI/UX in these changes
  - [ ] My changes have passed muster with Erin and Jen via an over-the-shoulder review, screenshots, etc.
  - [ ] Erin and Jen have not had an opportunity to provide feedback yet
 
###  Backend
   - [ ] I have considered error handling and notification to slack alert rooms via `LOG.error()`.
     #### Route Changes
     - [ ] My changes involve a new route changes to an existing route, so the who/what/when are logged using the usual logback mechanism
at the _start_ of the route, prior to any validation, as well as at the end of the route when work completes.
     ### Housekeeping Changes
     - [ ] My changes involve updates to housekeeping, and I am logging the who/what/when via logback.
  ### Client-side
   - [ ] My changes do not involve backend route change, but I have useful logging statements nonetheless.

## Analytics
 - [ ] I have discussed the analytics needs at both a platform and study level with Jen and instrumented code
 accordingly.
 - [x] There are no analytics requirements for these changes

###  Backend
   - [ ] These changes alter fundamental auth filter logic
   - [ ] These changes introduce a new backend API route that is behind an auth filter
   - [ ] These changes introduce a new backend API that _does not require auth_
   - [ ] These changes alter auth0 internals (rules, hosted login, configuration, etc.)
   - [ ] These changes expose new information to elastic and I have verified that PHI is not being exposed to the public
    
  ### Browser
  - [ ] These changes alter what we store in browser storage
  
  ### Mobile
  - [ ] TBD 
 
 ## Performance and Stability
  - [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, and performance bottlenecks and written a brief summary in this PR
  - [ ] I'm unsure about stability, performance, fault tolerance, and graceful degradation.  Please help!
  
## Testing
 - [ ] I have written automated positive tests
 - [ ] I have written automated negative tests
 - [x] I have written zero automated tests but have poked around locally to verify proper functionality
 - [ ] The jira ticket has acceptance criteria and Kiara has in the information she needs to test my changes
 
## Release
 - [x] These changes require no special release procedures--just code!
 - [ ] Releasing these changes requires special handling
   - [ ] I have documented the special procedures in the release plan document
   - [ ] The special procedures require minimal downtime
   - [ ] The special procedures require unavoidable, extended downtime for:
     - [ ] Participant UX
     - [ ] Study staff

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java [hunk 1]
```diff
@@ -145,6 +146,8 @@ public Object handle(Request request, Response response, UserRegistrationPayload
                 // if the invitation is revoked or there's already an account for the user, error out.
                 if (invitation.isVoid()) {
                     throw new DDPException("Invalid invitation " + invitationGuid + " for user " + auth0UserId);
+                } else if (invitation.isAccepted()) {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java:149]: I think we should also check if invitation has already been accepted? CC @zyme

**Comment by yufengwng** [comment]
You can now try some of the age-up stuff locally. But for accepting invitations, we'll need to wait for [this fix](https://github.com/broadinstitute/ddp-angular/pull/22) on the angular side before it'll work locally.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java (hunk 1 above)
Comment (zyme) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java:149]: Yep!  Thanks.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java [hunk 2]
```diff
@@ -158,19 +161,20 @@ public Object handle(Request request, Response response, UserRegistrationPayload
...
-                        LOG.info("User {} has been associated  with auth0 id {}", user.getGuid(), auth0UserId.get());
-
                         if (numRows != 1) {
                             throw new DDPException("Updated " + numRows + " for " + auth0UserId.get());
                         }
+                        LOG.info("User {} has been associated with auth0 id {}", user.getGuid(), auth0UserId.get());
+
                         invitationDao.updateAcceptedAt(TimestampUtil.now(), invitationGuid);
 
-                        // todo arz when DDP-4222 lands, trigger events here
```
Comment (zyme) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java:173]: Woo hoo!

File: pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java [hunk 3]
```diff
@@ -71,7 +71,8 @@ public void testVerificationOfGoodInvitation() throws Exception {
                 HttpStatus.SC_OK, response.returnResponse().getStatusLine().getStatusCode());
 
         TransactionWrapper.useTxn(handle -> {
-            InvitationDto requeriedInvitation = handle.attach(InvitationDao.class).findByInvitationGuid(invitation.getGuid()).get();
+            InvitationDto requeriedInvitation = handle.attach(InvitationDao.class)
```
Comment (zyme) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java:74]: It's probably a lot of work to setup test infrastructure to also boot housekeeping, block for a bit, and then look for evidence that the event has been processed, but I wonder if there's something that we can do with a mock/spy to verify that `EventService.processAllActionsForEventSignal()` has been called for the given signal?

**Review by zyme — DISMISSED** [review]

File: pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java:74]: I guess one way might be to indirectly test it by explicitly adding event configuration that hooks into the signal trigger type, something simply and verifiable. So in a way we check that it indeed triggers "downstream" events.

**Comment by yufengwng** [comment]
please run tests

File: pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/src/test/java/org/broadinstitute/ddp/route/VerifyInvitationRouteTest.java:74]: Updated the tests to do this.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/service/AgeUpServiceTest.java [hunk 4]
```diff
@@ -133,6 +167,9 @@ public void testRunAgeUpCheck_reachedAgeOfMajority() {
             assertEquals(user1.getId(), candidates.get(0).getParticipantUserId());
             assertTrue(candidates.get(0).hasInitiatedPrep());
 
+            assertTrue("activity instance should have been made read-only", handle.attach(JdbiActivityInstance.class)
```
Comment (zyme) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/service/AgeUpServiceTest.java:170]: Cool check!

**Review by zyme — APPROVED** [review]

**Pull request merged**